### ISSUE TITLE
HTTP::Server hangs the client with a null body

### ISSUE THREAD

**Issue by RX14** [manastech/crystal#1253]
The following code causes clients to hang when connecting to the server:
```crystal
require "http/server"

HTTP::Server.new(8080) do
    HTTP::Response.new 403
end.listen
```

however using an empty string for the body instead of `nil` works fine.

CURL logs:
```
$ curl -vS http://localhost:8080/
*   Trying ::1...
* Connected to localhost (::1) port 8080 (#0)
> GET / HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.43.0
> Accept: */*
> 
< HTTP/1.1 403 Forbidden
< Connection: keep-alive
* no chunk, no close, no size. Assume close to signal end
< 
```

**Comment by lbguilherme** [comment]
This has been fixed by #1315 (merged).

**Comment by RX14** [comment]
Thanks!
